frameworks/cassandra/README.md

```diff
@@ -1 +1,646 @@
+
+<a name="disaster-recovery"></a>
+# Disaster Recovery
+
+<a name="backup"></a>
+## Backup
+
+You can backup an entire cluster's data and schema to Amazon S3 using the `backup-s3` plan. This plan requires the following parameters to run:
+- `SNAPSHOT_NAME`: the name of this snapshot. Snapshots for individual nodes will be stored as gzipped tarballs with the name `$SNAPSHOT_NAME-<POD_INDEX>`.
+- `CASSANDRA_KEYSPACES`: the Cassandra keyspaces to backup. The entire keyspace, as well as its schema, will be backed up for each keyspace specified.
+- `AWS_ACCESS_KEY_ID`: the access key ID for the AWS IAM user running this backup.
+- `AWS_SECRET_ACCESS_KEY`: the secret access key for the AWS IAM user running this backup.
+- `AWS_REGION`: the region of the S3 bucket being used to store this backup.
+- `S3_BUCKET_NAME`: the name of the S3 bucket to store this backup in.
+
+Make sure that you provision your nodes with enough disk space to perform a backup. The backups that Apache Cassandra are stored on disk before being uploaded to S3, and will take up as much space as the data currently in the tables, so you'll need half of your total available space to be free to backup every keyspace at once.
+
+As noted in the documentation for the [backup/restore strategy configuration option](#backup-restore-strategy), it is possible to run transfers to S3 either in serial or in parallel, but care must be taken not to exceed any throughput limits you may have in your cluster. Throughput depends on a variety of factors, including uplink speed, proximity to region where the backups are being uploaded and downloaded, and the performance of the underlying storage infrastructure. You should perform periodic tests in your local environment to understand what you can expect from S3.
+
+You can configure whether snapshots are created and uploaded in serial, the default, or in parallel (see [backup/restore strategy](#backup-restore-strategy)), but the serial backup/restore strategy is recommended.
+
+You can initiate this plan from the command line:
+```
+SNAPSHOT_NAME=<my_snapshot>
+CASSANDRA_KEYSPACES="\"space1 space2\""
+AWS_ACCESS_KEY_ID=<my_access_key_id>
+AWS_SECRET_ACCESS_KEY=<my_secret_access_key>
+AWS_REGION=us-west-2
+S3_BUCKET_NAME=backups
+dcos cassandra plan start backup-s3 "SNAPSHOT_NAME=$SNAPSHOT_NAME,CASSANDRA_KEYSPACES=$CASSANDRA_KEYSPACES,AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID,AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY,AWS_REGION=$AWS_REGION,S3_BUCKET_NAME=$S3_BUCKET_NAME"
+```
+
+If you're backing up multiple keyspaces, you must wrap them in escaped quotation marks (`\"`) for the backup to run successfully. If the `CASSANDRA_KEYSPACES` parameter isn't supplied, then every keyspace in your cluster will be backed up.
+
+**IMPORTANT**: To ensure that sensitive information such as your AWS secret access key remains secure, make sure that you've set the `core.dcos_url` configuration property in the DC/OS CLI to an HTTPS URL.
```

Comment:
"To ensure that sensitive information, such as your AWS secret access key, remains secure..."